dart-impl/mpi/src/dart_communication.c

```diff
@@ -220,13 +225,15 @@ dart__mpi__get_basic(
 {
   if (num_reqs) *num_reqs = 0;
 
-  if (team_data->unitid == team_unit_id.id) {
+  if (team_data->unitid == team_unit_id.id &&
+      num_local_memcpy++ < NUM_CONSECUTIVE_MEMCPY) {
     // use direct memcpy if we are on the same unit
     memcpy(dest, seginfo->selfbaseptr + offset,
         nelem * dart__mpi__datatype_sizeof(dtype));
     DART_LOG_DEBUG("dart_get: memcpy nelem:%zu "
         "source (coll.): offset:%lu -> dest: %p",
         nelem, offset, dest);
+    num_local_memcpy = 0;
     return DART_OK;
   }
 
```

Comment:
`get_shared_mem` and `put_shared_mem` also call `memcpy`, are they also effected by this?

Reply:
They do not, as that doesn't depend on progress triggered locally.